File: platform-api/src/foam.ts
import * as foam from '@foam-ai/node-opentelemetry';
import { FOAM_API_KEY, IS_PRODUCTION } from './config';
import { HttpInstrumentation } from '@opentelemetry/instrumentation-http';
import type { InstrumentationBase } from '@opentelemetry/instrumentation';

foam.init({
  serviceName: 'platform-api',
  isProduction: IS_PRODUCTION,
  apiKey: FOAM_API_KEY,
  additionalInstrumentations: [
    new HttpInstrumentation({}) as unknown as InstrumentationBase,
  ],
});

File: ai-agent-runtime/src/foam.py
import foam
from config import FOAM_API_KEY, IS_PRODUCTION

foam.init(
    service_name="ai-agent-runtime",
    api_key=FOAM_API_KEY,
    environment="production" if IS_PRODUCTION else "development",
    # Enable automatic instrumentation for FastAPI, asyncpg, httpx
    auto_instrument=True,
)